### DISCUSSION TITLE
What's the best/cheapest Ethernet for RP2040?

### DISCUSSION THREAD

**Discussion by LeSuedois** [earlephilhower/arduino-pico#549] [Q&A]
Hello everybody,
I need Ethernet for a project, what type of ethernet can you advice for the RP2040?

The price and stock is an argument since I have to make like 100 devices...

Thank you for your advices,
Simon

**Comment by JAndrassy** [comment]
with Arduino library there are only two options: one of the Wiznet W5x00 chips series with the Arduino Ethernet library or ENC28j60 with my EthernetENC library.

**Reply by LeSuedois** [reply]
Hello @JAndrassy thank you for your reply, I found one of these in my stock and give it a try!

**Comment by khoih-prog** [comment]
Try the combined [**W5500-EVB-Pico**](https://docs.wiznet.io/Product/iEthernet/W5500/w5500-evb-pico/), [**W5100S-EVB-Pico**](https://docs.wiznet.io/Product/iEthernet/W5100S/w5100s-evb-pico/) or [**WIZnet Ethernet HAT**](https://docs.wiznet.io/Product/Open-Source-Hardware/wiznet_ethernet_hat/)

They are supported by [**Ethernet_Generic**](https://github.com/khoih-prog/Ethernet_Generic), [**EthernetWebServer**](https://github.com/khoih-prog/EthernetWebServer), [**EthernetWebServer_SSL**](https://github.com/khoih-prog/EthernetWebServer_SSL), etc. libraries


![image](https://user-images.githubusercontent.com/57012152/165033940-90295b42-ea58-40dc-bdde-41ad71c139c2.png)

![image](https://user-images.githubusercontent.com/57012152/165033992-9d5c0878-1933-490e-931e-8d34935b58be.png)

![image](https://user-images.githubusercontent.com/57012152/165035051-203ec95d-af5f-4c23-8b1a-168c9dc3f94e.png)

**Reply by LeSuedois** [reply]
Hello @khoih-prog thank you for your reply. This cards are clearly my first choice! Sad to say that the european Distributor is out of stock and can't announce any lead time ;(

P.s. sorry for the issue I created recently on one of your repos. I swear I will never create a issue after 10 pm... My bad..
